pkg/blob/azure_keychain.go

```diff
@@ -0,0 +1,35 @@
+package blob
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/sas"
+)
+
+var (
+	claim      = "https://storage.azure.com/.default"
+	apiHeader  = "x-ms-version"
+	apiVersion = sas.Version
+)
+
+type azKeychain struct{}
+
+func (a azKeychain) Resolve(url string) (string, map[string]string, error) {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return "", nil, err
+	}
+
+	tk, err := cred.GetToken(context.Background(), policy.TokenRequestOptions{Claims: claim})
```

Comment:
should set Scopes thant than Claims
`policy.TokenRequestOptions{Scopes: []string{xxx}}`